snuba/query/processors/custom_function.py

```diff
@@ -52,12 +73,15 @@ def __init__(
         dataset_schema: ColumnSet,
         name: str,
         signature: Sequence[Tuple[str, ParamType]],
-        body: str,
+        body: Expression,
     ) -> None:
         self.__dataset_schema = dataset_schema
         self.__function_name = name
-        self.__param_names, param_types = zip(*signature)
-        self.__body = parse_clickhouse_function(body)
+        self.__param_names = []
+        param_types = []
+        if len(signature) > 0:
+            self.__param_names, param_types = zip(*signature)
```

Comment:
nit: I don't think you need this change anymore.

Reply:
No I still do. failure_rate_processor has an empty signature since all its values are hardcoded, and this call fails with empty arrays.

Reply:
mh, ok good point.